app/src/main/java/com/ustwo/boilerplate/base/BaseActivity.java

```diff
@@ -5,17 +5,18 @@
 import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
 import android.support.v7.app.AppCompatActivity;
+import butterknife.ButterKnife;
 import com.ustwo.boilerplate.application.ApplicationComponent;
 import com.ustwo.boilerplate.application.BoilerplateApplication;
 
-public abstract class BaseActivity<V extends BaseView, C extends BaseComponent>
-        extends AppCompatActivity {
+public abstract class BaseActivity<V extends BaseView> extends AppCompatActivity {
 
     @Override
     protected void onCreate(@Nullable final Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        createComponentAndInject();
         setContentView(getLayoutId());
+        ButterKnife.bind(this);
+        onInject();
         onInitialize();
         getPresenter().onViewAttached(getPresenterView());
```

Comment:
Not sure it's work wrapping since it's just a one-liner. If it becomes more complex we should re-consider.